xds/src/main/java/io/grpc/xds/CommonXdsClient.java

```diff
@@ -244,12 +250,30 @@ protected void handleLdsResponse(String versionInfo, List<Any> resources, String
       }
       ldsUpdates.put(listenerName, update);
     }
+
+    // code to process serverSideListeners to generate ListenerUpdate’s
+    Map<String, ListenerUpdate> listenerUpdates = new HashMap<>();
```

Comment:
We should first eliminate one of the Listener/LDS watchers before trying to put server side logic into this XdsClient implementation.